### ISSUE TITLE
How to train the batch normalization?

### ISSUE THREAD

**Issue by John1231983** [matterport/Mask_RCNN#315]
I found that you have frozen the batch normalization in your model.py

Because I am working on a different task, likes medical image. Hence, I have to train the batch normalization. First, I will use the small size of input likes 128x128 to enable training with a batch size of 16. How could I modify the model.py?

Second question: After completed the training batch normalization with the batch size of 16 and image size of 128x128, I will freeze the batch size (as you did) and train the network with the size of 512x512 with checkpoint from the size of 128x128. Which layer should I ignore when I use my pre-trained model?

**Comment by AlphaQi** [comment]
USE GN
https://link.zhihu.com/?target=https%3A//arxiv.org/abs/1803.08494

**Comment by John1231983** [comment]
Him have you try it? I have used it in pytorch but it is not good. How about in keras?

**Comment by waleedka** [comment]
@John1231983 Batch normalization is disabled by hard-coding training=False in the `BatchNorm` class in `model.py`. You just need to undo that to re-enable batch normalization.

**Comment by baoruxiao** [comment]
Hi @waleedka, if I want to sync the bacthnorm among multi-gpus, do your parallel_model.py has already supported it? Thanks

**Comment by USTClj** [comment]
Also has the issue:  sync the bacthnorm among multi-gpus.
